python_ta/cfg/visitor.py

```diff
@@ -128,6 +129,20 @@ def visit_functiondef(self, func: nodes.FunctionDef) -> None:
         self._current_cfg.start.add_statement(func.args)
         func.cfg_block = self._current_cfg.start
 
+        if z3 is not None:
+            # Parse types
+            expr = ExprWrapper(func)
+            z3_vars_list = expr.parse_function_def(func)
+            # Confirm that type list is given
+            if isinstance(z3_vars_list, list):
+                self._current_cfg._z3_vars.update(
+                    {z3_var.decl().name(): z3_var for z3_var in z3_vars_list}
```

Comment:
The dictionary's keys should use the `func.args` argument names, not the names of the variables. In principle these could be different, and it's good to stick with the source of truth where possible.

That said, your current implementation of `parse_function_def` skips parameters if there's no type annotation, or if the type annotation can't be inferred, so the returned list could be of a different length than `func.args`. You can fix this by instead having the function return a dictionary mapping the function argument names to `z3` variables, and then simply passing that return value directly to `self._current_cfg._z3_vars.update`.